quixstreams/sources/community/file/origins/base.py

```diff
@@ -0,0 +1,53 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, BinaryIO, Iterable, Union
+
+__all__ = ("FileOrigin",)
+
+
+@dataclass
+class FileOrigin:
+    """
+    An interface for interacting with a file-based client.
+
+    Provides methods for navigating folders and retrieving/opening raw files.
+    """
+
+    _client: Any
+    _credentials: Union[dict, str]
+    root_location: Union[str, Path]
+
+    @property
+    @abstractmethod
+    def client(self): ...
```

Comment:
Do we need this property and does it have to be _public_? Maybe we can get away with only `_client` attribute and on class implementations. The base class does not need to be aware of all the additional shit its subclasses might need to do its job :)

Reply:
I definitely see your argument and considered similar...though I figured that it seems likely that only "local" will not use one? 

As such, I concluded it seemed more reasonable to account for the majority rather than a special case. Wdyt?

Reply:
I don't fully agree but it's not a blocker either 😉 So feel free to resolve or you can ask Daniil what he thinks.

One alternative would be to introduce an intermediate class like `ExternalOrigin(Origin)` and then `S3Origin(ExternalOrigin)`. I didn't think about it before, I could do the same thing for destinations 🤔

Reply:
Another thought that crossed my mind 🙂 wassnt trying to over engineer it since the only thing different is client, but it would be a small change. Seems like a good middle ground 👍